pkg/registry/apis/provisioning/resources/id.go

```diff
@@ -92,24 +92,12 @@ func appendHashSuffix(hashKey, repositoryName string) func(string) string {
 }
 
 // Will pick a name based on the hashed repository and path
```

Comment:
I tried to simplify this a lot by extracting some logic into `safepath` package.